lib/rubocop/cop/rails/find_by.rb

```diff
@@ -45,6 +46,15 @@ def autocorrect(node)
             corrector.replace(first_loc, '')
           end
         end
+
+        private
+
+        def where_with_method_as_param?(node)
+          where = node.children.find { |el| el.method?(:where) }
```

Comment:
I'm not entirely sure, but this should break *some* case. Might be worth figuring out which, and add a test case for it. 🙂